### PR TITLE
Specialization auth handler

### PR DESCRIPTION
**Pull request by mattthiessendev**
- feat: create SpecializationAuthorizationHandler

### CODE REVIEW

File: webapi/Auth/SpecializationAuthorizationHandler.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    SpecializationRepository specializationRepository,
+    ChatSessionRepository chatSessionRepository
+) : AuthorizationHandler<SpecializationRequirement, IContextValueAccessor>
+{
+    private const string GROUP_CLAIM_TYPE = "groups";
+
+    protected override async Task HandleRequirementAsync(
+        AuthorizationHandlerContext context,
+        SpecializationRequirement requirement,
+        IContextValueAccessor resource
```
Comment (mattthiessendev) [inline webapi/Auth/SpecializationAuthorizationHandler.cs:23]: info: this param is named `resource` to reflect the [pattern](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.authorization.authorizationhandler-2?view=aspnetcore-8.0) of `AuthorizationHandler<TRequirement, TResource>`.

Comment (mattthiessendev) [reply webapi/Auth/SpecializationAuthorizationHandler.cs:23]: `AuthorizationHandler.Resource` houses this parameter.

File: webapi/Auth/SpecializationAuthorizationHandler.test.cs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    private const string SPECIALIZATION_ID = "b10723df-8312-41b6-be80-703f0af2e192";
+    private const string GROUP_ID = "e67ecc45-b456-4109-9546-c177800d0186";
+    private SpecializationAuthorizationHandler? handler;
+    private AuthorizationHandlerContext? context;
+    private ChatSession? chatSession;
+    private Mock<IContextValueAccessor>? resource;
+    private Mock<IStorageContext<ChatSession>>? chatSessionContext;
+    private Mock<IStorageContext<Specialization>>? specializationContext;
+
+    private AuthorizationHandlerContext BuildAuthorizationContext(IEnumerable<Claim> claims)
```
Comment (mattthiessendev) [inline webapi/Auth/SpecializationAuthorizationHandler.test.cs:26]: thought: there is an opportunity to pull this into a test utils class. If we did something like this we may want to consider pulling these test files into their own test structure so we don't clutter the source code directories.

**Comment by wdavidcamden** [comment]
Testing Teams to Github Comment

**Pull request merged**